Patch no longer works but it should give you a clear idea what needs to be changed

diff --git a/app/javascript/mastodon/actions/search.js b/app/javascript/mastodon/actions/search.js
index 7c06670eb..ee7bc1657 100644
--- a/app/javascript/mastodon/actions/search.js
+++ b/app/javascript/mastodon/actions/search.js
@@ -37,7 +37,7 @@ export function submitSearch() {
       params: {
         q: value,
         resolve: true,
-        limit: 5,
+        limit: 20,
       },
     }).then(response => {
       if (response.data.accounts) {
diff --git a/app/models/status.rb b/app/models/status.rb
index fb9bbc9a9..f7718a14f 100644
--- a/app/models/status.rb
+++ b/app/models/status.rb
@@ -268,6 +268,13 @@ class Status < ApplicationRecord
   after_create :set_poll_id
 
   class << self
+    def search_for(term, limit = 20, offset)
+      pattern = sanitize_sql_like(term)
+      pattern = "#{pattern}"
+      Status.unscoped {
+        Status.where('tsv @@ plainto_tsquery(?)', pattern).where(visibility: [:public]).order(updated_at: :desc).limit(limit).offset(offset)
+      }
+    end
     def selectable_visibilities
       visibilities.keys - %w(direct limited)
     end
diff --git a/app/services/search_service.rb b/app/services/search_service.rb
index e0da61dac..f3aaae9e1 100644
--- a/app/services/search_service.rb
+++ b/app/services/search_service.rb
@@ -16,6 +16,7 @@ class SearchService < BaseService
         results[:accounts] = perform_accounts_search! if account_searchable?
         results[:statuses] = perform_statuses_search! if full_text_searchable?
         results[:hashtags] = perform_hashtags_search! if hashtag_searchable?
+       results[:statuses] = Status.search_for(query.gsub(/\A#/, ''), limit, offset) unless query.start_with?('@') or query.start_with?('#')
       end
     end
   end
